File: src/lib/scala/Slate.Common/src/main/scala/slate/common/lex/LexResult.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2016 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common.lex

import slate.common.Strings


/**
  * Result of lexical parsing. This could represent all the tokens or just a subset
  * @param success  : whether or not the lex parse was valid
  * @param message  : a message if there were errors
  * @param tokens   : the tokens parsed
  * @param total    : the total number of tokens parsed
  * @param isSubset : whether this represents a subset of tokens or all tokens
  * @param ex       : exception if failure
  */
case class LexResult(
                      success : Boolean      = false,
                      message : String       = null,
                      tokens  : List[Token] = null,
                      total   : Int          = 0,
                      isSubset: Boolean      = true,
                      ex      : Exception    = null
                    )
{
  def toStringDetail():String =
  {
    val text = s"$success, $message, total=$total, isSubset=$isSubset"
    var tokenDetail = ""
    for(token <- tokens)
    {
      tokenDetail += token.toStringDetail() + Strings.newline
    }
    text + Strings.newline + tokenDetail
  }
}

File: src/common/common/src/main/kotlin/kiit/common/lex/LexResult.kt
/**
 *  <kiit_header>
 * url: www.kiit.dev
 * git: www.github.com/slatekit/kiit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github 
 *  </kiit_header>
 */

package kiit.common.lex

import kiit.common.newline

/**
 * Result of lexical parsing. This could represent all the tokens or just a subset
 * @param success : whether or not the lex parse was valid
 * @param message : a message if there were errors
 * @param tokens : the tokens parsed
 * @param total : the total number of tokens parsed
 * @param isSubset : whether this represents a subset of tokens or all tokens
 * @param ex : exception if failure
 */
data class LexResult(
    val success: Boolean,
    val message: String,
    val tokens: List<Token>,
    val total: Int,
    val isSubset: Boolean,
    val ex: Exception?
) {
    fun toStringDetail(): String {
        val text = "$success, $message, total=$total, isSubset=$isSubset"
        val tokenDetail = tokens.fold("", { s, t -> s + t.toStringDetail() + newline })
        return text + newline + tokenDetail
    }
}
